include/fastrtps/qos/SubscriptionMatchedStatus.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file SubscriptionMatchedStatus.h
+*/
+
+#ifndef _SUBSCRIPTION_MATCHED_STATUS_H_
+#define _SUBSCRIPTION_MATCHED_STATUS_H_
+
+#include <cstdint>
+
+#include <fastrtps/rtps/common/InstanceHandle.h>
+#include <fastrtps/rtps/common/MatchingInfo.h>
+
+namespace eprosima{
+namespace fastrtps{
+
+//! @brief A structure storing the subscription status
+struct SubscriptionMatchedStatus
+{
+	//! @brief Constructor
+    SubscriptionMatchedStatus()
+        : total_count()
+        , total_count_change()
+        , current_count()
+        , current_count_change()
+        , last_publication_handle()
+        , status(rtps::MATCHED_MATCHING)
+    {} 
+
+    //! @brief Destructor
+    ~SubscriptionMatchedStatus()
+    {}
+
+	//! @brief Total cumulative count the concerned reader discovered a match with a writer
+	//! @details It found a writer for te same topic with a requested QoS that is compatible with that offered by the reader
+	uint32_t total_count = 0;
```

Comment:
It would be better to use int32_t instead as RTI does.

Reply:
Changed